src/Config.ts

```diff
@@ -621,19 +625,33 @@ export class Config implements ConfigParams, ParserConfig {
         throw new ConfigValueEmpty('currencySymbol')
       }
     })
-    this.validateNumberToBeAtLeast(this.maxColumns, 'maxColumns', 1)
+    validateNumberToBeAtLeast(this.maxColumns, 'maxColumns', 1)
 
-    this.checkIfParametersNotInConflict(
+    configCheckIfParametersNotInConflict(
       {value: this.decimalSeparator, name: 'decimalSeparator'},
       {value: this.functionArgSeparator, name: 'functionArgSeparator'},
       {value: this.thousandSeparator, name: 'thousandSeparator'}
     )
   }
 
-  public getConfig(): ConfigParams { //TODO: avoid pollution
-    return this
+  public getConfig(): ConfigParams {
+    const ret = {}
+    for (const key in Config.defaultConfig) {
+      const val = this[key as ConfigParamsList]
+      if (Array.isArray(val)) {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+        // @ts-ignore
```

Comment:
Do we need those because of missing type for `ret`? If so, an utilty type `Partial<T>` may help. We may need to cast it to full ConfigParams later but I see that we do it anyway

```ts
const ret: Partial<ConfigParams> = {}

// ...

return ret as ConfigParams
```

Reply:
I don't think there is an easy way to type this.

`this[key]` is a heterogenously typed object, and `ret[key]` is also heterogenously typed, and IMO typescript has a hard time understanding that those will be of the same type.